Azure CosmosDB for NoSQL (#4714)
SumoLogic/sumologic-documentation: docs/integrations/microsoft-azure/azure-cosmos-db.md

* Gremlin
* Table

:::note

> This app has been tested with MongoDB, Cassandra and NoSQL database account types.

Reviewer: ```suggestion
This app is tested with MongoDB, Cassandra, and NoSQL database account types.
```

:::

##### Enabling Microsoft Defender for Cloud

For Security events, make sure you enable [Microsoft Defender for Cloud](https://learn.microsoft.com/en-us/azure/defender-for-cloud/defender-for-databases-enable-cosmos-protections?tabs=azure-portal). In Defender Plans Settings page toggle the Databases status under Cloud Workload Protection section and select the resource types.

<img src={useBaseUrl('img/integrations/microsoft-azure/Microsoft-Cloud-Defender-Edit-Settings.png')} alt="Edit Settings" style={{border: '1px solid gray'}} width="800" />

<img src={useBaseUrl('img/integrations/microsoft-azure/Microsoft-Cloud-Defender-Plans-CosmosDB.png')} alt="Cloud Defender Plans" style={{border: '1px solid gray'}} width="800" />

## Installing the Azure CosmosDB for NoSQL app

This section provides instructions on how to install the Azure CosmosDB for NoSQL app, and shows examples of each of the preconfigured dashboards you can use to analyze your data.

import AppInstall2 from '../../reuse/apps/app-install-v2.md';

<AppInstall2/>

## Viewing the Azure CosmosDB for NoSQL dashboards

import ViewDashboards from '../../reuse/apps/view-dashboards.md';

<ViewDashboards/>

### Overview

> 

Reviewer: Dashboard info needs to be added.

1. [Configure an HTTP Source](/docs/send-data/collect-from-other-data-sources/azure-monitoring/collect-metrics-azure-monitor/#step-1-configure-an-http-source).
1. [Configure and deploy the ARM Template](/docs/send-data/collect-from-other-data-sources/azure-monitoring/collect-metrics-azure-monitor/#step-2-configure-azure-resources-using-arm-template).
1. [Export metrics to Event Hub](/docs/send-data/collect-from-other-data-sources/azure-monitoring/collect-metrics-azure-monitor/#step-3-export-metrics-for-a-particular-resource-to-event-hub). Perform the following steps for each Azure Cosmos DB account that you want to monitor.
   1. Choose **Stream to an event hub** as destination.
   1. Select all the metrics under **Metrics** section.
   1. Use the Event Hub namespace created by the ARM template in the previous step. You can create a new Event Hub or use the one created by the ARM template. You can use the default policy `RootManageSharedAccessKey` as the policy name.<br/><img src={useBaseUrl('img/integrations/microsoft-azure/azurecosmosdb-metrics.png')} alt="Azure Cosmos DB Diagnostic Settings for metrics" style={{border: '1px solid gray'}} width="800" />

:::note

> Currently only Azure Cosmos DB for NoSQL database account type supports exporting metrics using diagnostic settings.

Reviewer: ```suggestion
Currently, only Azure Cosmos DB for NoSQL database account type supports exporting metrics using diagnostic settings.
```
